File: 257. Binary Tree Paths.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {string[]}
 */
let binaryTreePaths = function(root) {
    let output = []
    
    if (!root) {
        return output;
    }
    
    let find = function(root, nxt) {
        if (!root.left && !root.right) {
            output.push(nxt + root.val)
        }
        
        if (root.left) {
            find(root.left, nxt + root.val + "->")
        }
        
        if (root.right) {
            find(root.right, nxt + root.val + "->")
        }
    }
    
    find(root, "")
    return output
};

File: 257. Binary Tree Paths.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def binaryTreePaths(self, root):
        """
        :type root: TreeNode
        :rtype: List[str]
        """
        output = []
        
        def findPath(root, path):
            if root:
                if not root.left and not root.right:
                    output.append(path + str(root.val))
                elif not root:
                    return None
                else:
                    findPath(root.left, path + str(root.val) + "->")
                    findPath(root.right, path + str(root.val) + "->")
        
        findPath(root, "")
        
        return output
